Makefile

```diff
@@ -24,7 +24,8 @@ fmt:
 	@bin/fmt.sh
 
 .PHONY: build
-build:	
+build: kubeconfig
+	@bin/install-prereqs.sh
```

Comment:
install-prereqs is using `go get` which doesn't cache well. This means the build will be slowed down by quite a bit.
How about this: take a look at prow presubmit scripts. You should see a setup section there. Let's create `setup` task in `make` that does similar set-upy things (you may even split out those scripts out of prow-presubmit). Then leave build be just the bazel build.

Reply:
I see. So there are the following steps in the order of dependency:
1. add kubeconfig (for testing)
2. bazel build/test (to fetch dependencies)
3. bin/init.sh (to vendorize dependencies)
4. install prereqs (gometalinter mostly)
5. bin/check.sh for linting (to run linters on vendorized go-built code)

As long as you don't add any back dependency, we're good. Feel free to choose where to make the cut or how to cache steps.

Reply:
Perhaps updating the document may be a good choice, what do you think @kyessenov

Reply:
I'm out for a few days, but yeah, will get to updating docs once back.